src/wiki/plugins/linknetwork/models.py

```diff
@@ -0,0 +1,129 @@
+"""Define a model for storing which article links to which other article
+
+Also define the function that stores links in that model, and connect it to the
+ArticleRevision.post_save signal, so that links are always as up-to-date as
+article revisions.
+
+"""
+from urllib.parse import urljoin
+from urllib.parse import urlparse
+from xml.etree import ElementTree as ET
+
+from django.db import models
+from django.urls import resolve
+from django.urls.exceptions import Resolver404
+from django.utils.translation import gettext_lazy as _
+from wiki import models as wiki_models
+from wiki.core.exceptions import NoRootURL
+from wiki.core.markdown import article_markdown
+from wiki.decorators import which_article
+
+
+__all__ = ["InternalLink", "store_links"]
+
+
+class InternalLink(models.Model):
+    """This model describes links between articles."""
+
+    from_article = models.ForeignKey(
+        "wiki.Article",
+        on_delete=models.CASCADE,
+        verbose_name=_("from_article"),
```

Comment:
The `verbose_name` properties and `help_text` can be used to add a bit more clarity about what this field is, also the `to_article` field. They are a bit hard to grasp :) Would be nice to have it written out.